packages/story-editor/src/components/hotlinkModal/useHotlinkModal.js

```diff
@@ -107,6 +109,66 @@ function useHotlinkModal({
         return;
       }
 
+      if ('image' === hotlinkInfo?.type && requiredImgDimensions) {
+        const proxiedUrl = needsProxy
+          ? getProxiedUrl({ needsProxy }, link)
+          : link;
+
+        const dimensions = await getImageDimensions(proxiedUrl);
+        const { height: suppliedHeight, width: suppliedWidth } = dimensions;
+        const { height: requiredHeight, width: requiredWidth } =
+          requiredImgDimensions;
+        if (
+          requiredHeight &&
+          requiredHeight !== suppliedHeight &&
+          requiredWidth &&
+          requiredWidth !== suppliedWidth
+        ) {
+          const message = sprintf(
+            /* translators: 1: supplied width. 2: supplied height. 3: desired width. 4: desired height */
+            __(
+              'Image dimensions (%1$dx%2$dpx) do not match required image dimensions (%3$dx%4$dpx).',
+              'web-stories'
+            ),
+            suppliedWidth,
+            suppliedHeight,
+            requiredWidth,
+            requiredHeight
+          );
+          setErrorMsg(message);
+          return;
+        }
+        if (requiredHeight && requiredHeight !== suppliedHeight) {
+          const message = sprintf(
+            /* translators: 1: supplied height. 2: desired height */
+            __(
+              'Image height (%1$dpx) does not match required image height (%2$dpx).',
+              'web-stories'
+            ),
+            suppliedHeight,
+            requiredHeight
+          );
+          setErrorMsg(message);
+          return;
+        }
+        if (requiredWidth && requiredWidth !== suppliedWidth) {
+          const message = sprintf(
+            /* translators: 1: supplied width. 2: desired width */
+            __(
+              'Image width (%1$dpx) does not match required image width (%2$dpx).',
+              'web-stories'
+            ),
+            suppliedWidth,
+            requiredWidth
```

Comment:
Note that my [suggestion](https://github.com/GoogleForCreators/web-stories-wp/pull/11775#discussion_r911786912) was deliberately written so that the characters `x` and `px` are not part of the translatable string but part of the placeholder instead.

This way translators will make less mistakes like accidentally messing up the placeholders or omitting the px or similar.

```suggestion
            /* translators: 1: image dimensions. 2: required dimensions. */
            __(
              'Image dimensions (%1$s) do not match required image dimensions (%2$s).',
              'web-stories'
            ),
            `${suppliedWidth}x${suppliedHeight}px`,
            `${requiredWidth}x${requiredHeight}px`
          );
          setErrorMsg(message);
          return;
        }
        if (requiredHeight && requiredHeight !== suppliedHeight) {
          const message = sprintf(
            /* translators: 1: supplied height. 2: required height. */
            __(
              'Image height (%1$s) does not match required image height (%2$s).',
              'web-stories'
            ),
            `${suppliedHeight}px`,
            `${requiredHeight}px`
          );
          setErrorMsg(message);
          return;
        }
        if (requiredWidth && requiredWidth !== suppliedWidth) {
          const message = sprintf(
            /* translators: 1: supplied width. 2: required width. */
            __(
              'Image width (%1$s) does not match required image width (%2$s).',
              'web-stories'
            ),
            `${suppliedWidth}px`,
            `${requiredWidth}px`
```

Reply:
My worry was that px would need to translated. As it is short for pixel, an english word. But if you say it better this way, then I will defer to you.